File: src/main/java/workshop/part3/UserActor.java
package workshop.part3;

import akka.actor.AbstractActor;
import akka.japi.pf.ReceiveBuilder;
import scala.PartialFunction;
import scala.runtime.BoxedUnit;
import workshop.common.userservice.UserCriminalRecord;
import workshop.common.userservice.UserService;

public class UserActor extends AbstractActor {

    private final UserService userService;

    public UserActor(UserService userService) {
        this.userService = userService;
    }

    @Override
    public PartialFunction<Object, BoxedUnit> receive() {
        return ReceiveBuilder.create()
            .match(CheckUser.class, m -> {
                UserCriminalRecord result = userService.vettUser(m.userId);
                sender().tell(new CheckUserResult(result), sender());
            })
            .build();
    }

    public static class CheckUserResult {
        public final UserCriminalRecord record;

        public CheckUserResult(UserCriminalRecord record) {
            this.record = record;
        }
    }

    public static class CheckUser {
        public final Integer userId;

        public CheckUser(Integer userId) {
            this.userId = userId;
        }
    }
}

File: src/main/kotlin/workshop/part4/UserActor.kt
package workshop.part4

import akka.actor.AbstractActor
import workshop.common.userservice.UserCriminalRecord
import workshop.common.userservice.UserService

class UserActor(private val userService: UserService) : AbstractActor() {

    override fun createReceive(): AbstractActor.Receive {
        return receiveBuilder()
                .match(CheckUser::class.java) { m ->
                    val result = userService.vettUser(m.userId)
                    sender().tell(CheckUserResult(result), sender())
                }
                .build()
    }

    class CheckUserResult(val record: UserCriminalRecord)

    class CheckUser(val userId: Int?)
}
